proxy/stats_get_prediction.go

```diff
@@ -0,0 +1,345 @@
+package proxy
+
+// Precaches statistics/get calls if the opening sequence of calls is detected
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const StatsGetWorkers = 5
+
+type StatsGetTask struct {
+	request  string
+	response chan *http.Response
+}
+
+type StatsGetPrediction struct {
+	enable         bool
+	GGStriveAPIURL string
+	loginPrefix    string
+	// TODO: Improve this.
+	// Counts up for env->login->stats/get, to kick off Async StatsGet predictive cache
+	statsGetPredictionCounter int
```

Comment:
Looks like this is used to keep track of API state probably better to do some sort of [enum](https://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go) to make it clear what this does.